diode-server/netbox/virtualization_wrappers.go

```diff
@@ -0,0 +1,1680 @@
+		if err != nil {
+			return nil, err
+		}
+		site.Tags = nil
+
+		if !actualSite.HasChanged() {
+			site = &DcimSite{
+				ID: actualSite.ID(),
+			}
+		}
+		vw.VirtualMachine.Site = site
+
+		vw.objectsToReconcile = append(vw.objectsToReconcile, siteObjectsToReconcile...)
+
+		clusterObjectsToReconcile, clusterErr := actualCluster.Patch(intendedCluster, intendedNestedObjects)
+		if clusterErr != nil {
+			return nil, clusterErr
+		}
+
+		cluster, err := copyData(actualCluster.Data().(*VirtualizationCluster))
+		if err != nil {
+			return nil, err
+		}
+		cluster.Tags = nil
+
+		if !actualCluster.HasChanged() {
+			cluster = &VirtualizationCluster{
+				ID: actualCluster.ID(),
+			}
+		}
+		vw.VirtualMachine.Cluster = cluster
+
+		vw.objectsToReconcile = append(vw.objectsToReconcile, clusterObjectsToReconcile...)
+
+		roleObjectsToReconcile, roleErr := actualRole.Patch(intendedRole, intendedNestedObjects)
+		if roleErr != nil {
+			return nil, roleErr
+		}
+
+		role, err := copyData(actualRole.Data().(*DcimDeviceRole))
+		if err != nil {
+			return nil, err
+		}
+		role.Tags = nil
+
+		if !actualRole.HasChanged() {
+			role = &DcimDeviceRole{
+				ID: actualRole.ID(),
+			}
+		}
+		vw.VirtualMachine.Role = role
+
+		vw.objectsToReconcile = append(vw.objectsToReconcile, roleObjectsToReconcile...)
+
+		if actualPlatform != nil {
+			platformObjectsToReconcile, platformErr := actualPlatform.Patch(intendedPlatform, intendedNestedObjects)
+			if platformErr != nil {
+				return nil, platformErr
+			}
+
+			platform, err := copyData(actualPlatform.Data().(*DcimPlatform))
+			if err != nil {
+				return nil, err
+			}
+			platform.Tags = nil
+
+			if !actualPlatform.HasChanged() {
+				platform = &DcimPlatform{
+					ID: actualPlatform.ID(),
+				}
+			}
+			vw.VirtualMachine.Platform = platform
+
+			vw.objectsToReconcile = append(vw.objectsToReconcile, platformObjectsToReconcile...)
+		}
+
+		if actualDevice != nil {
+			deviceObjectsToReconcile, deviceErr := actualDevice.Patch(intendedDevice, intendedNestedObjects)
+			if deviceErr != nil {
+				return nil, deviceErr
+			}
+
+			device, err := copyData(actualDevice.Data().(*DcimDevice))
+			if err != nil {
+				return nil, err
+			}
+			device.Tags = nil
+
+			if !actualDevice.HasChanged() {
+				device = &DcimDevice{
+					ID: actualDevice.ID(),
+				}
+			}
+			vw.VirtualMachine.Device = device
+
+			vw.objectsToReconcile = append(vw.objectsToReconcile, deviceObjectsToReconcile...)
+		}
+
+		tagsToMerge := mergeTags(vw.VirtualMachine.Tags, nil, intendedNestedObjects)
+
+		if len(tagsToMerge) > 0 {
+			vw.VirtualMachine.Tags = tagsToMerge
+		}
+
+		for _, t := range vw.VirtualMachine.Tags {
+			if t.ID == 0 {
+				vw.objectsToReconcile = append(vw.objectsToReconcile, &TagDataWrapper{Tag: t, hasParent: true})
+			}
+		}
+	}
+
+	if reconciliationRequired {
+		vw.hasChanged = true
+		vw.objectsToReconcile = append(vw.objectsToReconcile, vw)
+	}
+
```

Comment:
To ensure we don't have duplicated objects to reconcile, we should use `dedupObjectsToReconcile` func:
```suggestion

        dedupObjectsToReconcile, err := dedupObjectsToReconcile(vw.objectsToReconcile)
	if err != nil {
		return nil, err
	}
	vw.objectsToReconcile = dedupObjectsToReconcile

```